File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;

namespace Grid_MasterDetail_GetRelationName {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            // Load data
            this.suppliersTableAdapter.Fill(this.nwindDataSet.Suppliers);
            categoryProductsTableAdapter1.Fill(this.nwindDataSet.CategoryProducts);
            productsTableAdapter1.Fill(this.nwindDataSet.Products);

            // Create and customize a custom pattern View
            GridView customPatternView = new GridView(gridControl1);
            customPatternView.Columns.AddField("ProductName").VisibleIndex = 0;
            customPatternView.Columns.AddField("Discontinued").VisibleIndex = 1;
            customPatternView.OptionsView.ShowGroupPanel = false;
            // Associate the pattern View with the "CustomRelation" name
            gridControl1.LevelTree.Nodes.Add("CustomRelation", customPatternView);
        }

        // Provide a pattern View for the first master-detail relationship in the first master row
        private void advBandedGridView1_MasterRowGetRelationName(object sender, MasterRowGetRelationNameEventArgs e) {
            if (e.RowHandle == 0 && e.RelationIndex == 0)
                e.RelationName = "CustomRelation";
        }
        // Provide a display caption for the first master-detail relationship in the first master row
        private void advBandedGridView1_MasterRowGetRelationDisplayCaption(object sender, MasterRowGetRelationNameEventArgs e) {
            if (e.RowHandle == 0 && e.RelationIndex == 0)
                e.RelationName = "Custom Relation";
        }

    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid

Namespace Grid_MasterDetail_GetRelationName
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' Load data
			Me.suppliersTableAdapter.Fill(Me.nwindDataSet.Suppliers)
			categoryProductsTableAdapter1.Fill(Me.nwindDataSet.CategoryProducts)
			productsTableAdapter1.Fill(Me.nwindDataSet.Products)

			' Create and customize a custom pattern View
			Dim customPatternView As New GridView(gridControl1)
			customPatternView.Columns.AddField("ProductName").VisibleIndex = 0
			customPatternView.Columns.AddField("Discontinued").VisibleIndex = 1
			customPatternView.OptionsView.ShowGroupPanel = False
			' Associate the pattern View with the "CustomRelation" name
			gridControl1.LevelTree.Nodes.Add("CustomRelation", customPatternView)
		End Sub

		' Provide a pattern View for the first master-detail relationship in the first master row
		Private Sub advBandedGridView1_MasterRowGetRelationName(ByVal sender As Object, ByVal e As MasterRowGetRelationNameEventArgs) Handles advBandedGridView1.MasterRowGetRelationName
			If e.RowHandle = 0 AndAlso e.RelationIndex = 0 Then
				e.RelationName = "CustomRelation"
			End If
		End Sub
		' Provide a display caption for the first master-detail relationship in the first master row
		Private Sub advBandedGridView1_MasterRowGetRelationDisplayCaption(ByVal sender As Object, ByVal e As MasterRowGetRelationNameEventArgs) Handles advBandedGridView1.MasterRowGetRelationDisplayCaption
			If e.RowHandle = 0 AndAlso e.RelationIndex = 0 Then
				e.RelationName = "Custom Relation"
			End If
		End Sub

	End Class
End Namespace
